examples/js/exporters/GLTFExporter.js

```diff
@@ -433,27 +437,33 @@ THREE.GLTFExporter.prototype = {
 
 			}
 
-			var minMax = getMinMax( attribute );
-
-			var start = 0;
-			var count = attribute.count;
+			if ( start === undefined ) start = 0;
+			if ( count === undefined ) count = attribute.count;
 
 			// @TODO Indexed buffer geometry with drawRange not supported yet
 			if ( options.truncateDrawRange && geometry !== undefined && geometry.index === null ) {
 
-				start = geometry.drawRange.start;
-				count = geometry.drawRange.count !== Infinity ? geometry.drawRange.count : attribute.count;
+				var end = start + count;
+				var end2 = geometry.drawRange.count === Infinity
+						? attribute.count
+						: geometry.drawRange.start + geometry.drawRange.count;
+
+				start = Math.max( start, geometry.drawRange.start );
+				count = Math.min( end, end2 ) - start;
+
+				if ( count < 0 ) count = 0;
```

Comment:
quick thought, no changes needed for this PR: the code here would be a bit simpler if we didn't have to think about `.groups` and `.drawRange` separately... might be interesting to look at implementing `setDrawRange` differently:

```js
setDrawRange: function ( start, count ) {

	this.clearGroups();
	this.addGroup( start, count, 0 );

},
```

Reply:
Hm, for me `.groups` would be more like model-author-defined while `.drawRange` is runtime user-defined (maybe?) so better not to combine? For example, imagine user wanna narrow drawRange of multi-material mesh runtime for little time and then reset the range. Can't reset because original `.groups` info has been lost if we combine.

Reply:
Yeah, I think if we combined them it would be with the eventual goal of deprecating `setDrawRange()` entirely... it's not clear to me what happens when `.drawRange` and `.groups` are both set, so merging might decrease ambiguity. But just an idea, I don't have much preference on this.

Reply:
> it's not clear to me what happens when .drawRange and .groups are both set

You can try `mesh.geometry.setDrawRange(start, count)` on console on mmd example

https://threejs.org/examples/webgl_loader_mmd.html

This model has multi-material and groups.